### PR TITLE
d3JS implementation for substrait plan visualization

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR introduces the d3JS implementation to the `substrait-JS` package. The d3JS implementation will be a requirement for visualizing substrait plans in the substrait-JS CLI tool, and in the [`substrait-fiddle`](https://github.com/voltrondata/substrait-fiddle) project.

### CODE REVIEW

File: src/graph.ts [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+        '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" viewBox="0 0 16 16">' +
+        nodeIcon(d.type) +
+        "</svg>";
+      return str;
+    });
+
+  if(print_info){
+    // displaying node data on click
+    node.on("click", function (d:any) {
+      const node = document.getElementById("nodeData");
```
Comment (westonpace) [inline src/graph.ts]: The click handler should be a callback argument to `drawGraph`.  Then the library can choose how to handle the user selecting a node.  The code you have here now (e.g. `nodeData`) is specific to your use case I think and not a general part of "visualizing a plan".

**Review by westonpace — CHANGES_REQUESTED** [review]
We will need to add some unit tests.

File: src/graph.ts (hunk 1 above)
Comment (sanjibansg) [reply src/graph.ts]: Yes, that is correct. Thanks for the suggestion, now the handler is a callback argument to the `drawGraph()` function.

File: src/graph.ts [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    .html(function (d:Node) {
+      const str =
+        '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" viewBox="0 0 16 16">' +
+        nodeIcon(d.type) +
+        "</svg>";
+      return str;
+    });
+
+  if(print_info){
+    print_info(node, pre_nodes);
```
Comment (westonpace) [inline src/graph.ts]: I was thinking we would still hook it up to the `click` handler.  So something like...

```
if (print_info) {
    node.on("click", function (d:any) {
      const node = document.getElementById("nodeData");
      if(!node){
          throw new Error("Object nodeData was not created in DOM")
      }

      const nodeData = pre_nodes.get(d["currentTarget"]["__data__"]["name"]);
      if(!nodeData){
        throw new Error("Current target not found in nodes map")
      }
    print_info(node, nodeData);
  }
}
```

Comment (sanjibansg) [reply src/graph.ts]: made the change, thanks for the suggestion.

**Review by westonpace — APPROVED** [review]
We met externally and agreed to postpone unit tests until a CLI was added as that will make for an easier test API.

**Pull request merged**